android-components/components/feature/prompts/src/main/java/mozilla/components/feature/prompts/identitycredential/SelectAccountDialog.kt

```diff
@@ -0,0 +1,159 @@
+                    color = PhotonColors.DarkGrey90,
+                    letterSpacing = 0.15.sp,
+                ),
+            )
+        }
+
+        accounts.forEach { account ->
+            AccountItem(account = account, onClick = onAccountClick)
+        }
+
+        Spacer(modifier = Modifier.height(16.dp))
+    }
+}
+
+@Composable
+private fun AccountItem(
+    account: Account,
+    modifier: Modifier = Modifier,
+    onClick: (Account) -> Unit,
+) {
+    ListItem(
+        title = account.name,
+        description = account.email,
+        modifier = modifier,
+        onClick = { onClick(account) },
+    ) {
+        account.icon?.base64PngToBitmap()?.asImageBitmap()?.let { bitmap ->
+            Image(
+                bitmap = bitmap,
+                contentDescription = null,
+                contentScale = ContentScale.FillWidth,
+                modifier = Modifier
+                    .padding(horizontal = 16.dp)
+                    .size(32.dp),
+            )
+        } ?: Spacer(
+            Modifier
+                .padding(horizontal = 16.dp)
+                .width(32.dp),
+        )
+    }
+}
+
+@Composable
+@Preview(name = "Provider with no favicon")
+private fun AccountItemPreview() {
+    AccountItem(
+        modifier = Modifier.background(Color.White),
+        account = Account(
+            0,
+            "user@mozilla.com",
+            "User",
+            USER_PICTURE,
+        ),
+        onClick = {},
+    )
+}
+
+@Composable
+@Preview
+private fun SelectAccountDialogPreview() {
+    SelectAccountDialog(
+        provider = Provider(0, GOOGLE_FAVICON, "Google", "google.com"),
+        accounts = listOf(
+            Account(
+                0,
+                "user@mozilla.com",
+                "User",
+                USER_PICTURE,
+            ),
+            Account(
+                1,
+                "user2@mozilla.com",
+                "Google",
+                null,
+            ),
+        ),
+        modifier = Modifier.background(Color.White),
+    ) { }
+}
+
+@Suppress("MaxLineLength")
+private const val GOOGLE_FAVICON =
+    "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAARCAYAAADUryzEAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAAXNSR0IArs4c6QAAAARnQU1BAACxjwv8YQUAAAH+SURBVHgBhVO/b9NQEL5z7Bi1lWJUCYRYHBaWDh6SKhspAYnRoVO3SqxIbiUGEEIRQ1kbxB9AM7VISJiNoSUeI6DCYmSpt1ZUak2U0GLXuZ7dOnUsp73l+e593/343hkhZYePKqp/EhhAoLOrRkEEmwgd/mrd3PpmJvGYdPbvl1cJYQkuMQI085KwfP1Lxwl9Mb74Uyu/J4BFuMIEoKp/HCixHyXYf1BqEF2QuS13gPQ2P1OyQt//ta0CYgOBFApg7ob13R5ij9rX1P67uzuDv/k45ASBMHfLOmsxabvVipqOo78pNZls/Nu8Df7vAgRBrphFHmfhCPeEggdT8zvg2dNrk8/2Rsi1N12dx1OyyIjghgnUOCBrB5/TIAJhNYkZuSNwBT4vsiNlVrrEFJHf3UE6q7DtTfO5N4Jg5W3u1Um0pPBza+eeE4nIH8aHozvQ7M+4J3JQtOumO65kbaU33BfWwOK9IPN5dzYkRy1JntAYR3640tOSy8YatKJVLm3Mt/moJrAWEL7+sfDRCp3qJ13peYIxsft0SezPxjo5X19OFaMEGgOk/7mflK12OM5QXPlAB/mwDjjA+tarSXP4M1XWdXVCLrS7Xi8ryUhCsV9a7jx5sRbpkL4trz9eJBQMnlBLExncypHU7CxsOHEQx5WJ5j4WoyQiiE6SlLRTu5e/Z+DrlXsAAAAASUVORK5CYII="
```

Comment:
is this the raw byte data of sample images? If so, let's make sure these get cleaned up during the APK generation.

Reply:
Yes, exactly. 
I'm not sure how to make sure this is cleaned up during the APK generation, do you have any hint on that? 

Reply:
I don't unfortunately. If we use ProGuard, there should be some documentation somewhere with regards to the compose previews. 

One alternative would be to potentially make this a string resource inside the debug resource folder.

OR maybe we could take an existing vector, obtain its bitmap, and then transform it to bytecode?

Reply:
As discussed on other channels, according to [this stack overflow](https://stackoverflow.com/questions/69185826/how-can-i-prepare-a-jetpack-compose-app-to-release-what-about-the-preview-comp), this string should be removed from the release app. 
It's only referenced by a private Composable Preview function, which in turn is not referenced by anyone, so Proguard/R8 should remove it.